psalm.xml

```diff
@@ -56,7 +40,16 @@
         <MethodSignatureMismatch>
             <errorLevel type="suppress">
                 <file name="src/Behat/Testwork/EventDispatcher/TestworkEventDispatcherSymfonyLegacy.php" />
+                <file name="src/Behat/Testwork/EventDispatcher/TestworkEventDispatcherSymfony5.php" />
             </errorLevel>
         </MethodSignatureMismatch>
+
+        <!-- Psalm does not correctly understand the spread operator being applied on array_merge -->
```

Comment:
New error which is incorrectly thrown by the new version

Reply:
So, we have this code:
```
        return array_unique(
            array_merge(
                ...array_map(
                    function (Snippet $snippet) {
                        if (!$snippet instanceof ContextSnippet) {
                            return array();
                        }

                        return $snippet->getUsedClasses();
                    },
                    $this->snippets
                )
            )
        );
```

And PHPStan complains with:
`ERROR: NamedArgumentNotAllowed - src/Behat/Behat/Snippet/AggregateSnippet.php:116:20 - Method array_merge called with named unpacked array non-empty-array<array-key, array<array-key, string>> (array with string keys)`

So the issue seems to be that it understands that the unpack operator produces an `array-key` indexed array, which I think is wrong as it should be indexed by integers, as far as I know and this is what seems to be causing the issue. I tried several other approachs like saving the array result in another variable and type hinting it but could not get it to work. If you can think of a solution, suggestions are welcome

Reply:
`$this->snippets` is documented as `Snippet[]` which is a shortcut notation for `array<array-key, Snippet>`, not for `array<int, Snippet>`. So Psalm is actually right in reporting it right now.

Reply:
Not sure why, but I thought that `Snippet[]` was a `list` of snippets, i.e. the keys had to be ints. You never stop learning! I modified the code and now the error suppression is not needed